blog/2024-07-19-coordinator_async_setup.md

```diff
@@ -0,0 +1,118 @@
+---
+author: Josef Zweck
+authorURL: https://github.com/zweckj
+title: Set up your `DataUpdateCoordinator` with a setup method
+---
+
+In Home Assistant 2024.8, we are introducing the `_async_setup` method for the data update coordinator. 
+This method allows you to run asynchronous code to prepare your `DataUpdateCoordinator` instance
+or to load data that only needs to be loaded once.
+
+You can override `_async_setup` in your coordinator, and it will be automatically
+called during `coordinator.async_config_entry_first_refresh()`.
+It offers the same error handling as `_async_update_data` and will handle `ConfigEntryError`
+and `ConfigEntryAuthFailed` accordingly.
+
+## Example
```

Comment:
Instead of including these examples in the blog, update the dev docs for the coordinator and link to that page from the blog.

We shouldn't use the blog as documentation. It's just a way to announce the changes.

Reply:
I think a simple example is always included in the examples, I removed the second one. There is a PR already the docs already #2254, but I think it needs a bigger revamp, so let's do this in a follow up

Reply:
We should link to the docs page from this blog. In my opinion putting an example here is just duplicate work that may get outdated. We want to direct people to read the docs.

Reply:
I added a link to the documentation. 
I disagree. Because as dev on the blog, I want to read only the relevant updates with a snippet and not read the whole documentation. Even if the blog become outdated, that's no problem, since that's expected from a blog.
Imho we need a major revamp of the whole coordinator documentation, but I don't think that's in scope for this change.

Reply:
stripped down the example to a more lean one, so it's less documentation and more of a practical example.